renderdoc/driver/vulkan/vk_shaderdebug.cpp

```diff
@@ -42,6 +42,37 @@ RDOC_CONFIG(bool, Vulkan_Debug_DisableBufferDeviceAddress, false,
 RDOC_CONFIG(bool, Vulkan_Debug_ShaderDebugLogging, false,
             "Output verbose debug logging messages when debugging shaders.");
 
+namespace rdcspv
+{
+typedef rdcpair<uint32_t, rdcspv::Id> SwitchPairU32LiteralId;
+
+struct OpSwitch32
```

Comment:
This should be moved to `spirv_common.h` next to `OpExtInstHelper` and the other manual helper structs. This file is the only place currently using the OpSwitch helper but that doesn't guarantee it will be the case in the future.

At the same time, if you add a parallel `OpSwitch64` helper then you can use one or the other in the disassembly and debug cases instead of doing hand-decoding there, which would be much preferable.